### DISCUSSION TITLE
Elegant way to convert `OffsetDateTime` to `PrimitiveDateTime`

### DISCUSSION THREAD

**Discussion by liushuyu** [time-rs/time#472] [Question] [answered]
Is there an elegant way to convert `OffsetDateTime` to `PrimitiveDateTime` (given the timezone is already taken care of)?

Situation: an API from an external library accepts a `PrimitiveDateTime` but I currently own an `OffsetDateTime`.

Current hacky solution:

```rust
let input_time = datetime!(2000-01-01 0:00 +0);
let converted = PrimitiveDateTime(input_time.date(), input_time.time());
some_external_api(converted);
```

I am asking this question because there exists an elegant way to convert `PrimitiveDateTime` to `OffsetDateTime` (using `.assume_local` or `.assume_utc`).

**Comment by jhpratt** [comment, marked as answer]
That is currently the best solution. I have been extremely hesitant to add an API for this, as I feel it will be very error prone. I just had an idea that may work — `fn to_primitive_with_offset(self) -> (PrimitiveDateTime, UtcOffset)` as a method on `OffsetDateTime`. But I also just woke up, so we'll see how I feel about it later.

**Reply by jhpratt** [reply]
Coming back around to this. I have decided to _not_ implement this method. This is because it is possible for the stored `PrimitiveDateTime` to be outside the range of valid values (for external code), and returning it would violate the range validity. Effectively, it would be possible to obtain a `PrimitiveDateTime` that is larger than `PrimitiveDateTime::MAX` (and smaller than min).

While this could change depending on the internal representation, the fact that it's a possibility implicitly exposes some knowledge of the representation, which is not desirable.
